spharpy/classes/audio.py

```diff
@@ -0,0 +1,190 @@
+from pyfar import Signal
+from spharpy.spherical import renormalize, change_channel_convention
+import numpy as np
+
+
+class SphericalHarmonicSignal(Signal):
+    """Create audio object with spherical harmonics coefficients in time or
+    frequency domain.
+
+    Objects of this class contain spherical harmonics coefficients which are
+    directly convertible between time and frequency domain (equally spaced
+    samples and frequency bins), the channel conventions `acn` and `fuma`, as
+    well as the normalizations `n3d`, `sn3d`, or `maxn`. The default
+    parameters for `basis_type`, `normalization`, and `channel_convention`
+    corresponds to the AmbiX standard, see [#]_. The definition of the
+    spherical harmonics basis functions is based on the scipy convention which
+    includes the Condon-Shortley phase, [#]_, [#]_.
+
+
+    Parameters
+    ----------
+    data : ndarray, double
+        Raw data of the spherical harmonics signal in the time or
+        frequency domain. The memory layout of data is 'C'. E.g.
+        data of ``shape = (3, 4, 1024)`` has 3 channels with 4
+        spherical harmonic coefficients with 1024 samples or frequency
+        bins each. Time data is converted to ``float``. Frequency is
+        converted to ``complex`` and must be provided as single
+        sided spectra, i.e., for all frequencies between 0 Hz and
+        half the sampling rate.
+    sampling_rate : double
+        Sampling rate in Hz
+    n_max : int
+        Maximum spherical harmonic order. Has to match the number of
+        coefficients, such that the number of coefficients
+        :math:`>= (n_{max} + 1) ^ 2`.
```

Comment:
What is the idea of that? If it is the maximally possible SH order we could compute it from the size of data. If it is the transform order to be applied, it should be stated more clearly. It should also be mentioned that this must be smaller or equal to `np.sqrt(data.shape[-2]) - 1`.

Reply:
I think this is something for discussion during the weekly.